gcloud-java-bigquery/src/main/java/com/google/cloud/bigquery/TableInfo.java

```diff
@@ -63,6 +64,7 @@ public Table apply(TableInfo tableInfo) {
   private final Long creationTime;
   private final Long expirationTime;
   private final Long lastModifiedTime;
+  private final TimePartitioning timePartitioning;
```

Comment:
Why storing `TimePartitioning` if we only expose this option as a boolean?

Also I believe we should no expose it as just a boolean as we miss the chance to configure the [`expirationMs` field](timePartitioning.expirationMs). Also I have the feeling this time partitioning option will be extended to finer granularities in the future.

My preferred approach here is to create our own `TimePartitioning` class (that follows our code style: with `toPb` and `fromPb` methods, as well as convenience `builder` and `of` methods) and add that as a field to `TableInfo`. A `Builder timePartitioning(TimePartitioning)` setter should be added to the builder and a `TimePartitioning timePartitioning()` getter to `TableInfo`.

I can take care of applying such changes if you prefer.